lib/params.js

```diff
@@ -49,73 +56,54 @@ function getIndentationFromStylelintRules (rules) {
   }
 }
 
-function loadStylelintExtendConfig (params, config, configPath) {
-  if (isEmptyObject(config.extends)) {
-    return Promise.resolve(params)
-  }
-  return [].concat(config.extends).reduce(function (promise, extention) {
-    return promise.then(function (mergedParams) {
-      return augmentParams(mergedParams, extention, configPath)
-    })
-  }, Promise.resolve(params))
-}
-
-function augmentParams (originParams, extention, configPath) {
-  // same as the options in stylelint
-  var opts = {
-    configPath: resolveFrom(configPath, extention),
-    argv: false,
+function loadStylelintConfig (options, form) {
+  var stylelint
+  if (stylelintCache.has(options)) {
+    stylelint = stylelintCache.get(options)
+  } else {
+    const tailoredOptions = (options.rules)
+      ? { config: options }
+      : options
+    stylelint = createStylelint(tailoredOptions)
+    stylelintCache.set(options, stylelint)
   }
-  return cosmiconfig(extention, opts).then(function (stylelint) {
-    if (!stylelint) {
-      return Promise.resolve(originParams)
+  return stylelint.getConfigForFile(form, options.configFile).then(function (stylelint){
+    return stylelint && stylelint.config && stylelint.config.rules
+  }).catch(function (err) {
+    if(options.configFile) {
```

Comment:
This error occurs when we couldn't find `options.configFile` in `getConfigForFile`. Could it be into this `if` block?